enhancements/hypershift/topology-aware-scheduling/topology-aware-scheduling.md

```diff
@@ -0,0 +1,247 @@
+
+### Workflow Description
+
+The deployment and usage of NROP per-se, as a day2 operator,
+on HCP is similar to the deployment and usage on OCP. 
+No major changes are necessary on that front.
+
+### API Extensions
+
+Extend the [NUMAResourceOperator](https://github.com/openshift-kni/numaresources-operator/blob/0beeb62c7c0182ba5a2963ccd658bd0f6af44ebd/api/numaresourcesoperator/v1/numaresourcesoperator_types.go#L164) CR to contain NodeSelector filed.
+`spec.NodeGroup[].NodeSelector`
+User can choose the group of node by the NodeSelector and to have a unified API for OCP and HCP.
+
+For example:
+
+  OpenShift:
+
+  `spec.NodeGroup[].NodeSelector = "node-role.kubernetes.io/worker="`
+  
+  HyperShift:
+
+  `spec.NodeGroup[].NodeSelector = "hypershift.openshift.io/nodePool=hostedcluster02"`
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+This proposal aims for Hypershift / Hosted Control Planes alone. 
+
+#### Standalone Clusters
+
+* N/A
+
+#### Single-node Deployments or MicroShift
+
+* N/A
+
+### Implementation Details/Notes/Constraints
+
+#### Expose KubeletConfig to Hosted-Clusters
+
+At first, when a user associates a KubeletConfig's ConfigMap to the NodePool,
+the NodePool controller mirrors the ConfigMap to the control-plane namespace.
+Similar to: https://github.com/openshift/hypershift/pull/4150
+The KubeletConfig's ConfigMap NamespacedName convention, created by the NodePool controller is:
+`<control-plane-namespace>/<name+node-pool>`
+* `control-plane-namespace` = name of the control-plane namespace
+* `name` = name of the ConfigMap as provided by the user in the hosted-cluster namespace.
+* `node-pool` = name of the NodePool on which the ConfigMap was referenced.
+
+To make sure the final name is valid,
+it's provided to the https://github.com/openshift/hypershift/blob/ffa435f71e1a037174e29d31a2b80d569c940d57/support/util/route.go#L39 so that 
+```
+finalName := ShortenName(name, NodePoolName)
+``` 
+
+Then, HCCO picks it up from the control-plane namespace and copies it to the hosted-cluster.
+
+The KubeletConfig's ConfigMap NamespacedName convention, created by HCCO is:
+`openshift-config-managed/<name>`
+* `openshift-config-managed` = name of the namespace where all the configs managed by OCP reside.
+* `name` = name of the ConfigMap as provided by the NodePool controller in the control-plane namespace.
+
+In addition, KubeletConfig ConfigMap contains the following labels/annotations:
+
+Labels:
+* `hypershift.openshift.io/kubeletconfig-config` : `""`
+* `hypershift.openshift.io/nodePool` : `NodePool` API name where the KubeletConfig has referenced
+
+Annotations:
+* `hypershift.openshift.io/nodePool` : `NodePool` API name where the KubeletConfig has referenced
+
+When the KubeletConfig ConfigMap is deleted by the user,
+the hosted-cluster gets scraped and recreated again (which obviously deletes the ConfigMap as well).
```

Comment:
This is not clear. The hosted cluster does not get recreated by just adding/removing configmaps to the nodepool. Maybe I'm misunderstanding

Reply:
If a KubeletConfig config is being added to the cluster (encapsulated in ConfigMap ofc) , won't it trigger a recreation of the cluster so it contains the new configuration and data provided by KubeletConfig?

Reply:
it will trigger a re-rollout of the nodes, but the cluster remains unchanged

Reply:
Hmm, I need to do my own due diligence, but how re-rollout differ from recreation. 
For example when I checked, pods were gone/removed from the nodes, so I thought that other objects will get remove as well.

Reply:
Node re-rollout means that workers are recreated. However a cluster can exist without workers. All kubernetes API resources remain the same (so any configmaps/secrets/etc. remain). Pods are deleted when their nodes are deleted, so that's why you don't see pods.

Reply:
Yes @jparoly help me with that. Thank you both.
Any way I updated the enhancement to fit with the latest finding and comments.